### PR TITLE
maintenance fixes

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

**Comment by FFroehlich** [comment]
why are all the old commits showing up here?

**Comment by dweindl** [comment]
> why are all the old commits showing up here?

I guess you did not delete you local branch `fixes_ff`?

**Comment by FFroehlich** [comment]
true. but this used to work, probably no longer after using squash and merge?

**Comment by dweindl** [comment]
> true. but this used to work, probably no longer after using squash and merge?

Exactly. The last common ancestor is where the squashed branch was diverging from master. Best to delete also the local branch after merging... 
To fix this, do a rebase and push with `-f`.

File: .gitignore [hunk 1]
```diff
@@ -194,4 +195,6 @@ coverage_py.xml
 
 .idea/*
 
-python/examples/example_steadystate/model_steadystate_scaled/*
\ No newline at end of file
+python/examples/example_steadystate/model_steadystate_scaled/*
+
+python/sdist/*
```
Comment (dweindl) [inline .gitignore]: Let's do that more specific. There are files we want to track in this directory.

File: src/edata.cpp [hunk 2]
```diff
@@ -20,66 +22,93 @@ ExpData::ExpData(int nytrue, int nztrue, int nt, int nmaxevent)
...
+    for (int iz = 0; iz < nztrue; ++iz) {
+        std::normal_distribution<> e{0, sigma_z[iz]};
+        for (int ie = 0; ie < nmaxevent; ++ie) {
+            mz.at(iz + rdata.nztrue * ie) = rdata.z.at(iz + rdata.nz * ie) + e(gen);
+            sigmay.at(iz + rdata.nztrue * ie) = sigma_z[iz];
+        }
+    }
+}
 
+void ExpData::setTimepoints(const double *timepoints) {
```
[the remark this answers is not in the document]
Comment (FFroehlich) [reply src/edata.cpp]: Made this consistent with setObservedData et al, lets revisit this class in a seperate issue/PR.

File: src/edata.cpp [hunk 3]
```diff
@@ -20,66 +22,93 @@ ExpData::ExpData(int nytrue, int nztrue, int nt, int nmaxevent)
...
+    ts = rdata.ts;
+    
+    std::random_device rd{};
+    std::mt19937 gen{rd()};
+    
+    for (int iy = 0; iy < nytrue; ++iy) {
+        std::normal_distribution<> e{0, sigma_y[iy]};
+        for (int it = 0; it < nt; ++it) {
+            my.at(iy + rdata.nytrue * it) = rdata.y.at(iy + rdata.ny * it) + e(gen);
+            sigmay.at(iy + rdata.ny * it) = sigma_y[iy];
```
[the remark this answers is not in the document]
Comment (dweindl) [reply src/edata.cpp]: > currently sigma_y only specifies std per observable

Okay, should stated more clearly then and sigma_y and sigma_z be checked for correct size.

fill doesn't work, you are right. wrong slicing direction...

File: src/amici.cpp [hunk 4]
```diff
@@ -63,10 +63,10 @@ std::unique_ptr<ReturnData> runAmiciSimulation(Solver &solver, const ExpData *ed
...
         return rdata;
     }
-
+    
+    auto originalFixedParameters = model.getFixedParameters(); // to restore after simulation
+    auto originalTimepoints = model.getTimepoints();
     try{
-        std::vector<realtype> originalFixedParameters = model.getFixedParameters(); // to restore after simulation
-        std::vector<realtype> originalTimepoints = model.getTimepoints();
         if(edata) {
```
Comment (dweindl) [inline src/amici.cpp]: All this should also be outside `try`, not? 
``` 
        if(edata) {
            if(edata->fixedParameters.size()) {
                // fixed parameter in model are superseded by those provided in edata
                if(edata->fixedParameters.size() != (unsigned) model.nk())
                    throw AmiException("Number of fixed parameters (%d) in model does not match ExpData (%zd).",
                                       model.nk(), edata->fixedParameters.size());
                model.setFixedParameters(edata->fixedParameters);
            }
            if(edata->ts.size()) {
               // fixed parameter in model are superseded by those provided in edata
               model.setTimepoints(edata->ts);
            }
}
```

**Pull request merged**